src/ScreenCaptureKit/SCContentFilter.cs

```diff
@@ -0,0 +1,70 @@
+//
+// SCContentFilter.cs
+//
+// Authors:
+//	Alex Soto  <alexsoto@microsoft.com>
+//
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+namespace ScreenCaptureKit {
+
+#if NET
+	[UnsupportedOSPlatform ("ios")]
+	[UnsupportedOSPlatform ("tvos")]
+	[SupportedOSPlatform ("macos12.3")]
+	[SupportedOSPlatform ("maccatalyst15.4")]
+#else
+	[NoiOS]
+	[NoTV]
+	[NoWatch]
+	[Mac (12,3)]
+	[MacCatalyst (15,4)]
+#endif
+	public enum SCContentFilterOption {
+		Include,
+		Exclude,
+	}
+
+	public partial class SCContentFilter {
+
+		public SCContentFilter (SCDisplay display, SCWindow [] windows, SCContentFilterOption option) : base (NSObjectFlag.Empty)
+		{
+			switch (option) {
+			case SCContentFilterOption.Include:
+				Handle = InitWithDisplayIncludingWindows (display, windows);
+				break;
+			case SCContentFilterOption.Exclude:
+				Handle = InitWithDisplayExcludingWindows (display, windows);
+				break;
+			default:
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (option));
```

Comment:
The value isn't null, so I think AOORE is more appropriate:

```suggestion
				ObjCRuntime.ThrowHelper.ThrowArgumentOutOfRangeException (nameof (option));
```